mysql/azuremysql/azurecertfetcher.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2018 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package azuremysql provides connections to Azure Database for MySql.
+package azuremysql // import "gocloud.dev/mysql/azuremysql"
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"golang.org/x/net/context/ctxhttp"
+)
+
+type (
+
+	// AzureCertFetcher loads the CA certificates from remote endpoint or file.
+	AzureCertFetcher struct {
+		client *http.Client
+		// certLocation can be a remote endpoint or a file path
+		certLocation string
+		useHTTP      bool
+	}
+)
+
+// NewAzureCertFetcher constructs a new *AzureCertFetcher.
+// See https://docs.microsoft.com/en-us/azure/mysql/howto-configure-ssl.
+func NewAzureCertFetcher(caBundleLocation string) (*AzureCertFetcher, error) {
```

Comment:
Is there a reason to have a different location in Azure? Someone who wants it from a different location or a file should be able to provide their own implementation of an interface like `rds.CertPoolProvider`:

https://github.com/google/go-cloud/blob/7206007cdd0c778b597a28d615b2262c04cc7806/aws/rds/rds.go#L38-L41

Reply:
Can we possibly move this interface outside of aws/rds and make it more generic?

```
type CertPoolProvider interface {
	GetCertPool(context.Context) (*x509.CertPool, error)
}
```
This way, my azure cert provider can simply implement that contract?

Reply:
I think it's helpful to have two different interfaces to avoid potentially mixing types. Conceptually, these are returning two different certificate pools. Let's introduce `azure/azuredb` for this, since likely we'll want to reuse this certificate fetcher for PostgreSQL too.

Reply:
Ok, sounds good.